app/Main.hs

```diff
@@ -31,20 +22,48 @@ import Network.AWS.S3
        (ObjectCannedACL(OPublicRead), BucketName(BucketName),
         ObjectKey(ObjectKey), poACL, putObject)
 import Network.AWS.Data.Body (toBody)
+import Network.HTTP.Simple
+       (Request, parseRequest, addRequestHeader, getResponseStatus,
+        getResponseStatusCode, getResponseHeader)
+import Options.Applicative
+import System.Environment (setEnv, getEnv, lookupEnv)
+import System.Exit (ExitCode(..))
+import System.Directory (doesDirectoryExist)
+import System.IO.Temp (withSystemTempDirectory)
+
+import Stackage.Package.Update
+import Stackage.Package.Locations
+import Stackage.Package.IndexConduit
 
 run :: FilePath -> FilePath -> [String] -> IO ()
 run dir cmd args = do
-    putStrLn $ concat
-        [ "Running in "
-        , tshow dir
-        , ": "
-        , unwords $ map pack $ cmd : args
-        ]
-    withCheckedProcessCleanup
-        (proc cmd args) { cwd = Just dir }
-        (\Inherited Inherited Inherited -> return ())
+  putStrLn $
+    concat ["Running in ", tshow dir, ": ", unwords $ map pack $ cmd : args]
+  withCheckedProcessCleanup
+    (proc cmd args)
+    { cwd = Just dir
+    }
+    (\Inherited Inherited Inherited -> return ())
 
 
+-- | Similar to `run`, but will return the process' output as a `ByteString`.
+getOutput :: FilePath -> FilePath -> [String] -> IO ByteString
+getOutput dir cmd args = do
+  putStrLn $
+    concat ["Running in ", tshow dir, ": ", unwords $ map pack $ cmd : args]
+  (exitCode, out) <-
+    sourceProcessWithConsumer
+      (proc cmd args)
+      { cwd = Just dir
+      }
+      foldC
```

Comment:
Not that it should matter, but `sinkBuilder` or `sinkLazyBuilder` is likely to be more efficient.